sycl/include/sycl/ext/intel/experimental/esimd/detail/math_intrin.hpp

```diff
@@ -484,8 +484,18 @@ __esimd_dpas_inner(const __ESIMD_DNS::vector_type_t<T0, SZ> *src0,
   constexpr bool isPvc = SIMDSize == 16;
 
   constexpr bool
-      pvcHfDest = isPvc && std::is_same<RT, __ESIMD_EMU_DNS::half>::value,
-      pvcBfDest = isPvc && std::is_same<RT, short>::value,
+      pvcHfDest = isPvc && std::is_same_v<RT, unsigned short> &&
+                  src1_precision == __ESIMD_ENS::argument_type::FP16 &&
```

Comment:
FP16 looks inconsistent with the previous FP16->fp16 changes

Reply:
I forgot to do that change in this file as well.
If change FP16 with fp16 here, then also need to change "__ESIMD_ENS::argument_type" with "__ESIMD_NS::xmx::detail::dpas_argument_type"

Reply:
Thank you. I fixed this in this NFC fix: e828301a55c549ada6f4d6206fbaf3504fc9cd71